src/v8/eslint-dev-dependencies.ts

```diff
@@ -0,0 +1,104 @@
+import { existsSync, readFileSync, rmSync, writeFileSync } from "fs";
+
+import { PackageJson } from "../util/package-json.util";
+
+const adminEslintConfig = `import eslintConfigReact from "@comet/eslint-config/react.js";
+
+/** @type {import('eslint')} */
+const config = [
+    {
+        ignores: ["schema.json", "src/fragmentTypes.json", "dist/**", "src/**/*.generated.ts"],
+    },
+    ...eslintConfigReact,
+];
+
+export default config;`;
+const apiEslintConfig = `import eslintConfigNestJs from "@comet/eslint-config/nestjs.js";
+
+/** @type {import('eslint')} */
+const config = [
+    {
+        ignores: ["src/db/migrations/**", "dist/**", "src/**/*.generated.ts"],
+    },
+    ...eslintConfigNestJs,
+];
+
+export default config;`;
+const siteEslintConfig = `import eslintConfigNextJs from "@comet/eslint-config/nextjs.js";
+
+/** @type {import('eslint')} */
+const config = [
+    {
+        ignores: ["**/**/*.generated.ts", "dist/**", "lang/**", "lang-compiled/**", "lang-extracted/**", ".next/**", "public/**"],
+    },
+    ...eslintConfigNextJs,
+];
+
+export default config;`;
+
+export const stage = "before-install";
+
+/**
+ * This Updates script is doing following:
+ *
+ * - updates eslint to version 9.
+ * - creates new eslint.config.mjs file with default configuration
+ * - copies content from .eslint.json to new eslint.config.mjs as a comment
+ * - deletes old .eslint.json file
+ */
+export default async function updateEslint() {
+    if (existsSync("admin/package.json")) {
+        updateDevDependenciesInPackageJsons({ packageJsonPath: "admin/package.json" });
+        createNewFlatConfigurationFile({
+            workingDirectory: "admin/",
+            fileContent: adminEslintConfig,
+        });
+        deleteOldEslintRc({ workingDirectory: "admin/" });
+    }
+
+    if (existsSync("api/package.json")) {
+        updateDevDependenciesInPackageJsons({ packageJsonPath: "api/package.json" });
+        createNewFlatConfigurationFile({
+            workingDirectory: "api/",
+            fileContent: apiEslintConfig,
+        });
+        deleteOldEslintRc({ workingDirectory: "api/" });
+    }
+    if (existsSync("site/package.json")) {
```

Comment:
There are projects that have multiple sites (e.g., sites/site-a, sites/site-b). Should we change this script to check for all package.json files?

Reply:
we can try to do this.